PWGHF/TableProducer/candidateCreatorXic0Omegac0.cxx

```diff
@@ -1655,6 +1663,334 @@ struct HfCandidateCreatorXic0Omegac0 {
       }
     } // loop over LF Cascade-bachelor candidates
   }
+
+  template <o2::hf_centrality::CentralityEstimator centEstimator, int decayChannel, typename Coll, typename Hist>
+  void runOmegac0Xic0ToOmegaKaCreatorWithKFParticle(Coll const&,
+                                                    aod::BCsWithTimestamps const& /*bcWithTimeStamps*/,
+                                                    MyKfTracks const&,
+                                                    MyKfCascTable const&, KFCascadesLinked const&,
+                                                    aod::HfCascLf2Prongs const& candidates,
+                                                    Hist& hInvMassCharmBaryon,
+                                                    Hist& hFitterStatus,
+                                                    Hist& hCandidateCounter,
+                                                    Hist& hCascadesCounter)
+  {
+    for (const auto& cand : candidates) {
+      hCandidateCounter->Fill(1);
+
+      //----------------------check if the event is selected-----------------------------
+      auto collision = cand.collision_as<Coll>();
+      float centrality{-1.f};
+      const auto rejectionMask = hfEvSel.getHfCollisionRejectionMask<true, centEstimator, aod::BCsWithTimestamps>(collision, centrality, ccdb, registry);
+      if (rejectionMask != 0) {
+        /// at least one event selection not satisfied --> reject the candidate
+        continue;
+      }
+
+      //----------------------Set the magnetic field from ccdb-----------------------------
+      auto bc = collision.template bc_as<aod::BCsWithTimestamps>();
+      if (runNumber != bc.runNumber()) {
+        LOG(info) << ">>>>>>>>>>>> Current run number: " << runNumber;
+        initCCDB(bc, runNumber, ccdb, isRun2 ? ccdbPathGrp : ccdbPathGrpMag, lut, isRun2);
+        magneticField = o2::base::Propagator::Instance()->getNominalBz();
+        LOG(info) << ">>>>>>>>>>>> Magnetic field: " << magneticField;
+        runNumber = bc.runNumber();
+      }
+      KFParticle::SetField(magneticField);
+
+      // Retrieve skimmed cascade and pion tracks
+      auto cascAodElement = cand.cascade_as<aod::KFCascadesLinked>();
+      hCascadesCounter->Fill(0);
+      if (!cascAodElement.has_kfCascData()) {
+        continue;
+      }
+      auto casc = cascAodElement.kfCascData_as<MyKfCascTable>();
+      hCascadesCounter->Fill(1);
+
+      // convert KaonFromCharm&KaFromOmega&V0DauPos&V0DauNeg tracks into KFParticle object
+      auto trackKaFromCharm = cand.prong0_as<MyKfTracks>();
+      auto trackKaFromOmega = casc.bachelor_as<MyKfTracks>(); // Ka <- Omega track
+      auto trackV0DauPos = casc.posTrack_as<MyKfTracks>();    // V0 positive daughter track
+      auto trackV0DauNeg = casc.negTrack_as<MyKfTracks>();    // V0 negative daughter track
+      auto KaFromOmegaCharge = trackKaFromOmega.signed1Pt() > 0 ? +1 : -1;
```

Comment:
Why can't you use the `trackKaFromOmega.sign()` directly?

Reply:
@FDUEnrich did you check if you can use `trackKaFromOmega.sign()` directly instead of defining a new variable?